src/Components/DataHistoryExplorer/Internal/TimeSeriesTwinCallout/TimeSeriesTwinCallout.tsx

```diff
@@ -0,0 +1,235 @@
+                                        ModelledPropertyBuilderMode.PROPERTY_SELECT
+                                    }
+                                    propertyExpression={{
+                                        expression: timeSeriesTwinToEdit.twinPropertyName
+                                            ? `${PRIMARY_TWIN_NAME}.${timeSeriesTwinToEdit.twinPropertyName}`
+                                            : ''
+                                    }}
+                                    onChange={handleTwinPropertyChange}
+                                    isDisabled={!timeSeriesTwinToEdit.twinId}
+                                    required
+                                />
+                                {timeSeriesTwinToEdit.twinPropertyType &&
+                                    !isNumericType(
+                                        timeSeriesTwinToEdit.twinPropertyType
+                                    ) && (
+                                        <Toggle
+                                            styles={
+                                                classNames.subComponentStyles
+                                                    .typeCastToggle
+                                            }
+                                            defaultChecked={
+                                                timeSeriesTwinToEdit.chartProps
+                                                    .isTwinPropertyTypeCastedToNumber
+                                            }
+                                            label={t(
+                                                'dataHistoryExplorer.builder.timeSeriesTwin.propertyTypeCastLabel'
+                                            )}
+                                            onChange={
+                                                handleTwinPropertyCastingChange
+                                            }
+                                        />
+                                    )}
+                                <TextField
+                                    placeholder={t(
+                                        'widgets.dataHistory.form.timeSeries.labelPlaceholder'
+                                    )}
+                                    label={t(
+                                        'widgets.dataHistory.form.timeSeries.label'
+                                    )}
+                                    value={timeSeriesTwinToEdit?.label}
+                                    onChange={handleLabelChange}
+                                />
+                            </Stack>
+                            <Stack
+                                horizontal
+                                horizontalAlign={'end'}
+                                tokens={{ childrenGap: 8 }}
+                            >
+                                <PrimaryButton
+                                    text={t('add')}
```

Comment:
Text could be "Save" and potentially also have "Cancel" in the edit flow here?

<img width="306" alt="image" src="https://user-images.githubusercontent.com/4854357/210450584-120a7bfa-d471-4217-a835-145edcd51d14.png">